typescript-client/hatchet/clients/dispatcher/action-listener.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  DispatcherClient as PbDispatcherClient,
+  AssignedAction,
+  ActionEvent,
+} from '@protoc/dispatcher';
+
+import { ServerError, Status } from 'nice-grpc';
+import { ClientConfig } from '@clients/client/client-config';
+import sleep from '@util/sleep';
+import HatchetError from '@util/errors/hatchet-error';
+import { DispatcherClient } from './dispatcher-client';
+
+const DEFAULT_ACTION_LISTENER_RETRY_INTERVAL = 5; // seconds
+const DEFAULT_ACTION_LISTENER_RETRY_COUNT = 5;
+
+export class ActionListener {
+  config: ClientConfig;
+  client: PbDispatcherClient;
+  listener: AsyncIterable<AssignedAction>;
+  workerId: string;
+
+  constructor(client: DispatcherClient, listener: AsyncIterable<AssignedAction>, workerId: string) {
+    this.config = client.config;
+    this.client = client.client;
+    this.listener = listener;
+    this.workerId = workerId;
+  }
+
+  // actions
+  // parse_action_payload
+
+  actions = () =>
+    (async function* gen(client: ActionListener) {
```

Comment:
I am not sure this is the most optimal way of writing this. I havent worked with async generators in js before and am surprised how similar it is to python. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_async_iterator_and_async_iterable_protocols

@abelanger5 @steebchen did you have any thoughts on this?